extensions/wikidata/tests/src/org/openrefine/wikidata/qa/scrutinizers/DifferenceWithinScrutinizerTest.java

```diff
@@ -19,40 +47,78 @@ public EditScrutinizer getScrutinizer() {
+        Snak snak3 = Datamodel.makeValueSnak(maximumValuePID, maxValue);
+        List<SnakGroup> snakGroupList = makeSnakGroupList(snak1, snak2, snak3);
+        List<Statement> statementList = constraintParameterStatementList(entityIdValue, snakGroupList);
+
+        ConstraintFetcher fetcher = mock(ConstraintFetcher.class);
+        when(fetcher.getConstraintsByType(upperBoundPid, "Q21510854")).thenReturn(statementList);
+        when(fetcher.findValues(snakGroupList, "P2306")).thenReturn(Collections.singletonList(lowerBoundPid));
+        when(fetcher.findValues(snakGroupList, "P2313")).thenReturn(Collections.singletonList(minValue));
+        when(fetcher.findValues(snakGroupList, "P2312")).thenReturn(Collections.singletonList(maxValue));
+        setFetcher(fetcher);
+
         scrutinize(updateA);
         assertWarningsRaised(DifferenceWithinRangeScrutinizer.type);
     }
 
     @Test
     public void testNoIssue() {
         ItemIdValue idA = TestingData.existingId;
-        PropertyIdValue lowerBoundPid = MockConstraintFetcher.lowerBoundPid;
-        PropertyIdValue upperBoundPid = MockConstraintFetcher.differenceWithinRangePid;
-
         TimeValue lowerYear = new TimeValueImpl(2000, (byte)10, (byte)15, (byte)0, (byte)0, (byte)0, (byte)11, 0, 0, 0, TimeValue.CM_GREGORIAN_PRO);
         TimeValue upperYear = new TimeValueImpl(2020, (byte)10, (byte)15, (byte)0, (byte)0, (byte)0, (byte)11, 0, 0, 0, TimeValue.CM_GREGORIAN_PRO);
-
         ValueSnak value1 = Datamodel.makeValueSnak(lowerBoundPid, lowerYear);
         ValueSnak value2 = Datamodel.makeValueSnak(upperBoundPid, upperYear);
-
         Statement statement1 = new StatementImpl("P569", value1,idA);
         Statement statement2 = new StatementImpl("P570", value2,idA);
-
         ItemUpdate updateA = new ItemUpdateBuilder(idA).addStatement(statement1).addStatement(statement2).build();
+
+        Snak snak1 = Datamodel.makeValueSnak(propertyParameterPID, lowerBoundPid);
+        Snak snak2 = Datamodel.makeValueSnak(minimumValuePID, minValue);
+        Snak snak3 = Datamodel.makeValueSnak(maximumValuePID, maxValue);
+        List<SnakGroup> snakGroupList = makeSnakGroupList(snak1, snak2, snak3);
+        List<Statement> statementList = constraintParameterStatementList(entityIdValue, snakGroupList);
+
+        ConstraintFetcher fetcher = mock(ConstraintFetcher.class);
+        when(fetcher.getConstraintsByType(upperBoundPid, "Q21510854")).thenReturn(statementList);
+        when(fetcher.findValues(snakGroupList, "P2306")).thenReturn(Collections.singletonList(lowerBoundPid));
+        when(fetcher.findValues(snakGroupList, "P2313")).thenReturn(Collections.singletonList(minValue));
+        when(fetcher.findValues(snakGroupList, "P2312")).thenReturn(Collections.singletonList(maxValue));
+        setFetcher(fetcher);
+
         scrutinize(updateA);
         assertNoWarningRaised();
     }
+
+    private List<SnakGroup> makeSnakGroupList(Snak... snaks) {
```

Comment:
This could be pulled up in the `ScrutinizerTest` base class so that other tests can rely on it as well.